eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -586,53 +633,79 @@ def inits(self, matches):
                 result[init[1].children[0]] = int(init[0])
             else:
                 result[init[0].children[0]] = 1
-        return {'inits': result}
+        return {"inits": result}
 
     def param(self, matches):
         self.params.add(str(matches[0]))
         return Tree("param", matches)
 
     def model(self, matches):
+        rules = set()
         definitions = dict()
         regulation = None
         inits = collections.Counter()
         for match in matches:
             if type(match) == dict:
                 key, value = list(match.items())[0]
-                if key == 'rules':
-                    rules = set(value)
-                if key == 'inits':
-                    inits = value
-                if key == 'definitions':
-                    definitions = value
-                if key == 'regulation':
+                if key == "rules":
+                    rules.update(value)
+                if key == "inits":
+                    inits.update(value)
+                if key == "definitions":
+                    definitions.update(value)
+                if key == "regulation":
+                    if regulation:
+                        raise UnspecifiedParsingError("Multiple regulations")
+                    regulation = value
+            elif isinstance(match, Tree) and match.data == "sections":
+                if isinstance(match.children[0], Tree):
+                    continue
+                key, value = list(match.children[0].items())[0]
+                if key == "rules":
+                    rules.update(value)
+                if key == "inits":
+                    inits.update(value)
+                if key == "definitions":
+                    definitions.update(value)
+                if key == "regulation":
+                    if regulation:
+                        raise UnspecifiedParsingError("Multiple regulations")
```

Comment:
I believe this code can be simplified - the condtion influences only how `key, value` are created, the rest is the same in both cases.

Reply:
Right, I addressed it in commit a3eca18.